src/core/geo.js

```diff
@@ -104,14 +96,41 @@ var g = {
 
     }
   },
-  load: function(file, success) {
-    var ext = file.name.split('.')[1]
+
+  loadTextFormat: function (file, success) {
     var reader = new FileReader();
-    reader.onload = function(evt) {
+    reader.onload = function (evt) {
       success(JSON.parse(evt.target.result));
     };
     reader.readAsText(file);
   },
+
+  loadShapefile: function (file, success) {
+    /*
+    https://www.npmjs.com/package/shpjs
+    */
+
+    var reader = new FileReader();
+    reader.onload = function (evt) {
+      shp(evt.target.result).then(function (geojson) {
+        success(geojson);
+      });
+    };
+
+    reader.readAsArrayBuffer(file);
+  },
+
+  load: function (file, success) {
+    /* All load functions must call `success` handler with a GeoJSON feature */
```

Comment:
It's a *little* confusing reading the code with the success handler being passed down two levels. As long as that stack doesn't get any deeper (i.e. as long as the `load[type]file` methods don't do any more asynchronous operations) then we're probably OK. If it does get any deeper with another pass through, we may want to reorganize a bit. I think it's good for now, though.